src/App.tsx

```diff
@@ -8,55 +8,156 @@ import { PostsList } from './components/PostsList';
+  const [isCommentFormOpen, setIsCommentFormOpen] = useState(false);
+  const [isCommentsFetching, setIsCommentsFetching] = useState(false);
+  const [isPostlistVisible, setIsPostListVisible] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  const getUsersFromServer = async () => {
+    try {
+      const fetchedUsers = await getUsers();
+
+      setUsers(fetchedUsers);
+    } catch (error) {
+      setErrorType(ErrorTypes.USERS);
+    }
+  };
+
+  const getPostsFromServer = async (userId: number) => {
+    setIsLoaderVisible(true);
+    setIsPostListVisible(false);
+
+    try {
+      const fetchedPosts = await getPostsByUserId(userId);
+
+      setPosts(fetchedPosts);
+      setIsPostListVisible(true);
+    } catch (error) {
+      setErrorType(ErrorTypes.POSTS);
+      setIsPostListVisible(false);
+    } finally {
+      setIsLoaderVisible(false);
+    }
+  };
+
+  const getCommentsFromServer = async (postId: number) => {
+    setIsCommentsFetching(true);
+    setIsCommentsListHidden(false);
+
+    try {
+      const fetchedComments = await getPostComments(postId);
+
+      setComments(fetchedComments);
+      setIsPostListVisible(true);
+    } catch (error) {
+      setErrorType(ErrorTypes.COMMENTS);
+    } finally {
+      setIsCommentsFetching(false);
+    }
+  };
+
+  const handleOpenDetails = (post: Post) => {
+    setSelectedPost(post);
+    getCommentsFromServer(post.id);
+    setIsCommentFormOpen(false);
+  };
+
+  useEffect(() => {
+    getUsersFromServer();
+  }, []);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selected={selectedUser}
+                  onSelect={setSelectedUser}
+                  getUsersPosts={getPostsFromServer}
+                  setIsCommentsListHidden={setIsCommentsListHidden}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
 
-                <Loader />
+                {isLoaderVisible && (
+                  <Loader />
+                )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {(errorType === ErrorTypes.USERS
+                  || errorType === ErrorTypes.POSTS) && (
```

Comment:
```suggestion
                {[ErrorTypes.USERS, ErrorTypes.POSTS].includes(errorType) && (
```

Reply:
and the best solution is to move array into constant

Reply:
Fixed it 👍